src/frontend/src/optimizer/plan_node/utils.rs

```diff
@@ -62,13 +62,28 @@ impl TableCatalogBuilder {
         self.avoid_duplicate_col_name(&mut column_desc);
 
         self.columns.push(ColumnCatalog {
-            column_desc: column_desc.clone(),
+            column_desc,
             // All columns in internal table are invisible to batch query.
             is_hidden: false,
         });
         column_idx
     }
 
+    /// Extend the columns with column ids reset. This method does not change the column names and
+    /// does not perform duplicate column name check.
+    ///
+    /// Returns the indices of the extended columns.
+    pub fn extend_columns(&mut self, columns: &[ColumnCatalog]) -> Vec<usize> {
```

Comment:
I guess we still need dedup column name check here because we can not ensure the name is unique in the schema of plannode?  
If StreamSink has check it, I guess we need to do asseration here.